sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -513,6 +555,16 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     tableWithStats.copy(properties = getOriginalTableProperties(table))
   }
 
+  private def getLocationFromRawTable(rawTable: CatalogTable): Option[String] = {
+    rawTable.properties.get(TABLE_LOCATION).orElse {
+      // In older version of spark, we store the table location in storage properties with key
+      // `path`, instead of table properties with key `spark.sql.tableLocation`. We should
```

Comment:
Why do we need `spark.sql.tableLocation` instead of just relying on hive's location field and path in serde properties?